Tests/tests/DataStore2.spec.lua

```diff
@@ -79,6 +79,20 @@ return function()
 				expect(DataStore2(nonNilKey, fakePlayer):Get("badDefault")).to.equal("abc")
 			end)
 
+			it("should validate the data", function()
+				local dataStore = DataStore2(UUID(), fakePlayer)
+				local function testValidator(dataToValidate)
+					if dataToValidate == "yepp" then
+						return true
+					end
+					return false
+				end
+				dataStore:SetValidator(testValidator)
+				expect(dataStore.validator).to.be.a("function")
+				expect(dataStore:Set("nope")).to.throw()
+				expect(dataStore:Set("yepp")).to.equal(true)

```

Comment:
`Set` doesn't return anything, make sure you run the tests.

Reply:
Just noticed that